index.js

```diff
@@ -1,30 +1,53 @@
-const fs = require('fs');
+const axios = require('axios');
+var clientTransactionId = 1;
 
 exports.getStatus = async function(config) {
-  let weatherFile;
+  let weather = {
+    temperature: null,
+    dewPoint: null,
+    windSpeed: null,
+    windDirection: null,
+    windGust: null,
+    cloudCover: null,
+    humidity: null,
+    skyBrightness: null,
+    rainRate: null,
+    skyQuality: null,
+    skyTemperature: null,
+    starFwhm: null,
+    pressure: null
+  };
 
-  try {
-    weatherFile = await readFile(config.filePath);
-  } catch (error) {
-    throw new Error(`Cannot read file: ${config.filePath}`);
-  }
-
-  const lines = weatherFile.split('\n').map(line => line.split('='));
-
-  return lines.filter(line => line[0]).reduce(
-    (output, line) => ({
-      ...output,
-      [line[0]]: line[1],
-    }),
-    {},
-  );
+  let weatherProperties = Object.getOwnPropertyNames(weather);
+  await axios.all(weatherProperties.map(prop => getStat(config, prop.toLowerCase())))
```

Comment:
Might there be a better way to do this? If any one of the requests fail, then `.then` will never resolve. Is that desirable?  Or would it be better to resolve some of the weather values, even if one or two fail?

Reply:
Thanks for catching that!  My intention was indeed for the weather object to aggregated regardless of how many of the individual calls failed.  I fixed this by adding a catch block on the axios call within getStat.  The catch block returns null so getStat should always return some value.  The aggregation logic then checks for the erroneous null value and reacts appropriately.